blocks/init/src/Blocks/components/paragraph/components/paragraph-editor.js

```diff
@@ -31,15 +39,152 @@ export const ParagraphEditor = (attributes) => {
+
+				output = [...output, newElement];
+			}
+		}
+
+		return output;
+	}
+
+	/**
+	 * Process content and paste in into blocks smartly.
+	 *
+	 * @param {string} textFromClipboard Text retrieved from clipboard.
+	 */
+	const handlePaste = async (textFromClipboard) => {
+		// Split by newLine, discard empty lines
+		const parsedParagraphs = textFromClipboard.split('\n').filter((line) => line.length > 0);
+
+		// If something is selected replace it...
+		if (window.getSelection) {
+			const selection = window.getSelection();
+
+			let selectedRange = selection.getRangeAt(0);
+
+			// Process elements into nodes
+			const processedElements = processTag(parsedParagraphs[0]);
+
+			// Remove current text
+			if (!selectedRange.collapsed) {
+				selectedRange.deleteContents();
+			}
+
+			// Insert them one by one and keep spaces
+			for (const element of processedElements) {
+				selectedRange = selectedRange.cloneRange();
+				selectedRange.insertNode(element);
+				selectedRange.setStartAfter(element);
+				selectedRange.collapse(true);
+			}
+		}
+		// ...else append at the end
+		else {
+			setAttributes({ [`${componentName}Content`]: `${paragraphContent}${parsedParagraphs[0]}` });
+		}
+
+		for (const paragraph of parsedParagraphs.slice(1)) {
+
+			// Create a new block with the same attributes, but different content
+			const block = createBlock(`eightshift-boilerplate/${componentName}`, {
+				...attributes,
+				[`${componentName}Content`]: paragraph
+			});
+
+			const currentBlockClientId = wp.data.select('core/block-editor').getSelectedBlockClientId();
+			
+			const blockIndex = wp.data.select('core/block-editor').getBlockIndex(currentBlockClientId);
+
+			// Insert the block
+			wp.data.dispatch('core/block-editor').insertBlock(block, blockIndex + 1);
+		}
+	}
+
+	/**
+	 * Paste event handler.
+	 *
+	 * @param {Event} event Passed event parameters.
+	 */
+	const pasteInto = (event) => {
+		event.preventDefault();
+		const [copiedData, copiedHtml] = event.clipboardData.items;
+
+		// If HTML content is in clipboard, process that...
+		if (copiedHtml) {
+			copiedHtml.getAsString(async (inputText) => {
+				// Get all tags from the text
+				const allTags = [...inputText.matchAll(/<.+?>/g)];
+
+				// Filter out attributes and disallowed tags,
+				// make <p> and <br> elements give newline
+				for (const tag of allTags) {
+					const tagInnardsRegex = /(\S+)=["']?((?:.(?!["']?\s+(?:\S+)=|\s*\/?[>"']))+.)["']?/g;
+					const parsedTag = tag[0].replace(tagInnardsRegex, '');
+					const filteredTagName = parsedTag.replaceAll(/[<>/\s]+/g, '');
+
+					if (allowedTags.includes(filteredTagName)) {
+						inputText = inputText.replace(tag[0], parsedTag).replace(' >', '>');
+					}
+					else if ((filteredTagName === 'p' && !tag[0].includes('/')) || filteredTagName === 'br') {
+						inputText = inputText.replace(tag[0], '\n');
+					}
+					else {
+						inputText = inputText.replace(tag[0], '');
+					}
```

Comment:
```suggestion
					if (allowedTags.includes(filteredTagName)) {
						inputText = inputText.replace(tag[0], parsedTag).replace(' >', '>');
					} else if ((filteredTagName === 'p' && !tag[0].includes('/')) || filteredTagName === 'br') {
						inputText = inputText.replace(tag[0], '\n');
					} else {
						inputText = inputText.replace(tag[0], '');
					}
```

Just formatted to be a bit prettier.